### PR TITLE
Documentation practice

### PR DESCRIPTION
**Pull request by lyubomir-popov**
## Done

Based on @mtruj013's [pr](https://github.com/canonical/canonical.com/pull/905). I changed a few things so pushing to a separate pr so as to not break anything.

This is a bit different from the design, as I talked to Daniele since we did the pages and he expressed a preference for a denser treatment on documentation pages, and liked placing headings on the side and clamping the heading hierarchy to h4. 

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8002/
- Run through the following [QA steps](https://discourse.canonical.com/t/qa-steps/152)
- Check tab functionality on all 3 tabs

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-3737

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
@mtruj013 let's sync up about this on Monday to avoid merge conflicts. 
I'll talk to @juanruitina as well.

**Comment by mtruj013** [comment]
@juanruitina @lyubomir-popov what do you guys want to do about the side nav on medium and small screens? Since they're on the right in the new design the toggle button ends up on the bottom of the page. Should I hide it on smaller screens, decrease the cols the main content takes up (didn't look great when I tried it), or ? 
image

**Comment by lyubomir-popov** [comment]
> what do you guys want to do about the side nav on medium and small screens? Since they're on the right in the new design the toggle button ends up on the bottom of the page

To get to the bottom of the page, you've already gone past all headings linked to in the sidenav, so to me it can safely be hidden. the page isn't that long anyway

**Comment by mtruj013** [comment]
@lyubomir-popov oh agreed, I wasn't suggesting we keep it at the bottom of the page. I was asking if I should make room for it on the grid on med screens so that it isn't at the bottom. I think hiding it is the better option either way though, when I tried making the main content col-medium-4 to make space for the nav on the right it looked pretty bad

**Comment by juanruitina** [comment]
I agree with hiding the table of contents, there's little content anyway. Some other comments:

- [ ] Header needs some margin on top
- [ ] Tabs have orange accent and a white background, I understand they should be dark gray and transparent instead right? Same with table of contents: the colours need to be adjusted. @lyubomir-popov how are we doing this?
- [ ] "We're hiring" needs some padding on mobile (although I'm not sure how this block should behave there beyond that)
- [ ] There is some horizontal scrolling on mobile, dunno what element's the culprit tho.

**Comment by mtruj013** [comment]
Thanks all, I believe I've addressed all your comments. Lyubo's already filed https://github.com/canonical/vanilla-framework/issues/4816 for the z index issue on the sidenav. 

This is ready for all needed reviews, could y'all please take a look? @lyubomir-popov @juanruitina @ClementChaumel

**Comment by lyubomir-popov** [comment]
Thanks @mtruj013 looks good to me. Note to others - the demo seems to be showing a very outdated version, with orange bar under the selected tabs. Pull locally, works fine there:
image

**Comment by juanruitina** [comment]
LGTM :)

**Comment by juanruitina** [comment]
LGTM :)

**Comment by anthonydillon** [comment]
This is a comment for design on this one. On small screens the page header looks too close to the top of the page. With a large gap below. Making it look unbalanced: 


Maybe something to work on in the Vanilla typography + block structure @lyubomir-popov

**Comment by lyubomir-popov** [comment]
> This is a comment for design on this one. On small screens the page header looks too close to the top of the page. With a large gap below. Making it look unbalanced:

@anthonydillon this is an old bug, already fixed - the demo now shows it as it should be:

image

**Comment by lyubomir-popov** [comment]
@ClementChaumel this is UX+1 and awaiting reviews

File: templates/documentation/index.html [hunk 1]
```diff
@@ -43,71 +43,109 @@
...
+          <p>Documentation practitioners at Canonical are not just in the business of producing documentation &mdash; not even just the business of producing excellent documentation. They have a responsibility to advance the state of the art of documentation, and to advance documentation practice itself.</p>
+          <p>We work together towards this end, to define the practice of documentation. Our approach is critical, exploratory, collaborative and iterative.</p>
+          <h3 class="p-heading--4">We're hiring</h3>
+          <p>We're hiring technical authors to join multiple engineering teams, working on products across Canonical's portfolio.</p>
+          <p><a href="/careers/all?search=%22Technical+Author%22" class="p-button--positive">Join our team</a></p>
         </div>
-        <p><a href="https://ubuntu.com/blog/the-future-of-documentation-at-canonical">Read about our plan to put these pillars into place at Canonical&nbsp;&rsaquo;</a></p>
```
Comment (mtruj013) [inline templates/documentation/index.html:92]: This is one is needed as well

File: templates/documentation/work-and-careers.html [hunk 2]
```diff
@@ -39,82 +39,132 @@
...
-      <p>Our Technical Authors come from many backgrounds, including academic research, software engineering and journalism. What they have in common is that they are expert, effective communicators. They share an intense sense of technical curiosity, and a strong feeling for the perspective of a product's users. Their strength of imagination and qualities of empathy are crucial to their work.</p>
-      <p>Documentation is something that everyone contributes to; Technical Authors encourage and lead documentation work, and help guarantee its quality.</p>
-      <h2 class="question-heading p-heading--3" id="in-the-documentation-team" style="scroll-margin-top: 4.5rem;">In the documentation team</h2>
-      <div class="p-media-object">
-        <img src="https://assets.ubuntu.com/v1/f75f5612-tmihoc-photo.jpg" class="p-media-object__image is-round" alt="">
-        <div class="p-media-object__details">
-          <h3 class="p-media-object__title">Teodora Mihoc, Technical Author (Juju)</h3>
+      <hr class="p-rule">
+      <div class="row p-block">
+        <h2 class="col-3 col-medium-2 p-heading--4 question-heading" id="in-the-documentation-team">In the documentation team?</h2>
```
Comment (mtruj013) [inline templates/documentation/work-and-careers.html:77]:
```suggestion
        <h2 class="col-3 col-medium-2 p-heading--4 question-heading" id="in-the-documentation-team">In the documentation team</h2>
```

File: templates/documentation/work-and-careers.html [hunk 3]
```diff
@@ -39,82 +39,132 @@
...
+        <div class="col-6 col-medium-4">
+          <p>An interview to confirm our assessments and work out the next steps, based on your interests, skills and experience. This will help us arrange interviews with appropriate hiring managers in different teams.</p>
+        </div>
+        <div class="col-9 col-medium-6">
+          <hr class="p-rule">
+        </div>
+        <h3 class="col-3 col-medium-2 p-heading--5">Hiring manager interviews</h3>
+        <div class="col-6 col-medium-4">
+          <p>By this stage, we'd like to have you as a Technical Author at Canonical, and now it's a question of finding the right team or project for you, where both we and you feel that you'll be excited by the technology and in a strong position to make a valuable contribution.</p>
+          <p>That is indeed a lot of interviews. It's an in-depth process for an exacting role and we take it very seriously.</p>
```
Comment (mtruj013) [inline templates/documentation/work-and-careers.html:163]: I'm confused about this bit and I really wish we had an updated mock-up to work with. On the live site this is separated by an hr, and on the old mock-up it was handled like this:
image

But the new design doesn't seem to make a distinction between this section summary and and the "hiring manager interviews" portion. I think we should at least add an hr, what do you think?

**Review by mtruj013 — CHANGES_REQUESTED** [review]
Thanks @ClementChaumel! A couple of copy changes needed, all pretty minor. There also seems to be a bug with the tabs on the /beyond-canonical, it doesn't set the correct tab as active when clicked. Eg, the active/highlighted tab should be "Encounters and dialog" here
image

**Comment by ClementChaumel** [comment]
> Thanks @ClementChaumel! A couple of copy changes needed, all pretty minor. There also seems to be a bug with the tabs on the /beyond-canonical, it doesn't set the correct tab as active when clicked. Eg, the active/highlighted tab should be "Encounters and dialog" here image

I think it's due to the function using an intersection handler to highlight on scroll. It's also happening on live, I'll create an issue

**Review by mtruj013 — APPROVED** [review]
LGTM!

**Pull request merged**